packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButton.tsx

```diff
@@ -266,13 +275,15 @@ export const useSpinButton_unstable = (props: SpinButtonProps, ref: React.Ref<HT
 
     if (valueChanged || displayValueChanged) {
       onChange?.(e, { value: roundedValue, displayValue: newDisplayValue });
+    } else if (!valueChanged && newValue === null) {
+      onChange?.(e, { value: null });
```

Comment:
>{ value: null } [](http://example.com/codeflow?start=20&length=15)

Any thoughts on making the shape of the data returned consistent (e.g. make the object always have both `value` and `newDisplayValue`)? I think doing that could potentially allow consumers to simplify logic in their `onChange` callback